website/project/model.py

```diff
@@ -4004,16 +4008,51 @@ def approve(self, user):
         self.approval.approve(user)
         self.add_status_log(user, DraftRegistrationLog.APPROVED)
         self.approval.save()
+        self.checkin_files(save=True)
 
     def reject(self, user):
         self.approval.reject(user)
         self.add_status_log(user, DraftRegistrationLog.REJECTED)
         self.approval.save()
+        self.checkin_files(save=True)
 
     def add_status_log(self, user, action):
         log = DraftRegistrationLog(action=action, user=user, draft=self)
         log.save()
 
+    def get_metadata_files(self):
+        for q in ['q7', 'q16', 'q11', 'q13', 'q12', 'q19', 'q26']:
+            for file_info in self.registration_metadata[q]['value']['uploader']['extra']:
```

Comment:
What happens if one or more of these keys are missing, if a question is submitted without a file attached?